### ISSUE TITLE
Simple Dockerfile mode: provided Dockerfile is always skipped

### ISSUE THREAD

**Issue by dev-gaur** [eclipse/jkube#197]
Docs : https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin#simple-dockerfile-build

Reference : https://github.com/fabric8io/fabric8-maven-plugin/issues/1814

**Comment by rohanKanojia** [comment]
I think this issue is due to generators already adding opinionated `ImageConfiguration` as `resolvedImages` and `BuildConfiguration` is never null
https://github.com/eclipse/jkube/blob/1d5cda2fe84ea61ca8880c6ca569183d3f275ea9/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java#L820-L823

What should be the desired behavior here? In my opinion when simple `Dockerfile` is detected this `ImageConfiguration` should be replaced. But what if the user has provided XML configuration also for an image? What should be the precedence for configuration options? In my opinion, `Dockerfile` or resource fragments should be given most precedence, then XML configuration and then plugin's opinionated defaults.

**Comment by manusa** [comment]
> When only a single image should be built with a Dockerfile no XML configuration is needed at all. All what need to be done is to place a Dockerfile into the top-level module directory, alongside to pom.xml. You can still configure global aspects in the plugin configuration, but as soon as you add an <image> in the XML configuration, you need to configure also the build explicitly.


IMHO something which is configured should take precedence over the convention. This is also what's stated on the docs _"...but as soon as you add an <image> in the XML configuration..."_.

As a user, I may also face issues when something which I didn't configure takes precedence over what I'm expecting. e.g. User is completely unaware of the convention, so he places a Dockerfile in the module's root and configures XML to pick that Dockerfile but maybe add parts of the assembly from other directories, if the convention took precedence over the configured XML, the user wouldn't understand why the assembly was not working.

TL;DR Order of priorities:
1. Dockerfile in root directory
2. XML configuration
3. Generator defaults

**Comment by rohanKanojia** [comment]
Thanks for the feedback. Let me write a fix to check content of `resolvedImages` and replace it if `from` has one of the jkube opinionated images in it.

**Comment by manusa** [comment]
Blocked by #214
